src/ehrdata/_feature_types.py

```diff
@@ -0,0 +1,326 @@
+    `edata.var['feature_type']['feature1']='corrected_type'` or with :func:`~ehrdata.replace_feature_types`.
+    Be aware that not all features stored numerically are of `'numeric'` type, as categorical features might be stored in a numerically encoded format.
+    For example, a feature with values [0, 1, 2] might be a categorical feature with three categories. This is accounted for in the method, but it is
+    recommended to check the inferred types.
+
+    Args:
+        edata: Data object.
+        layer: The layer to use from the EHRData object. If `None`, the `X` field is used.
+        output: The output format. Choose between `'tree'`, `'dataframe'`, or `None`. If `'tree'`, the feature types will be printed to the console in a tree format.
+            If `'dataframe'`, a :class:`~pandas.DataFrame` with the feature types will be returned. If `None`, nothing will be returned.
+        verbose: Whether to print warnings for uncertain feature types.
+
+    Examples:
+        >>> import ehrdata as ed
+        >>> edata = ed.dt.mimic_2()
+        >>> ed.infer_feature_types(edata)
+    """
+    feature_types = {}
+    uncertain_features = []
+
+    X = edata.X if layer is None else edata.layers[layer]
+
+    if issparse(X):
+        X = X.toarray()
+
+    df = pd.DataFrame(X.reshape(-1, edata.shape[1]), columns=edata.var_names)
+
+    for feature in edata.var_names:
+        if (
+            FEATURE_TYPE_KEY in edata.var
+            and edata.var[FEATURE_TYPE_KEY][feature] is not None
+            and not pd.isna(edata.var[FEATURE_TYPE_KEY][feature])
+        ):
+            feature_types[feature] = edata.var[FEATURE_TYPE_KEY][feature]
+        else:
+            feature_types[feature], raise_warning = _detect_feature_type(df[feature])
+            if raise_warning:
+                uncertain_features.append(feature)
+
+    edata.var[FEATURE_TYPE_KEY] = pd.Series(feature_types)[edata.var_names]
+
+    if verbose:
+        logger.warning(
+            f"{'Features' if len(uncertain_features) > 1 else 'Feature'} {str(uncertain_features)[1:-1]} {'were' if len(uncertain_features) > 1 else 'was'} detected as categorical features stored numerically."
+            f"Please verify and correct using `ed.replace_feature_types` if necessary."
+        )
+
+        logger.info(
+            f"Stored feature types in edata.var['{FEATURE_TYPE_KEY}']."
+            f" Please verify and adjust if necessary using `ed.replace_feature_types`."
+        )
+
+    if output == "tree":
+        feature_type_overview(edata)
+    elif output == "dataframe":
+        return edata.var[FEATURE_TYPE_KEY].to_frame()
+    elif output is not None:
+        err_msg = f"Output format {output} not recognized. Choose between 'tree', 'dataframe', or None."
+        raise ValueError(err_msg)
+
+
+# TODO: this function is a different flavor of inferring feature types. We should decide on a single implementation in the future.
```

Comment:
Probably the 5th time I'm writing that comment. I'll now just assume that you'll solve that later with the ehrapy PR. Trust me, I'll annoy you about this haha.